reactive/zeppelin.py

```diff
@@ -47,3 +62,58 @@ def report_blocked():
 @when_not('spark.ready')
 def report_waiting(spark):
     hookenv.status_set('waiting', 'Waiting for Apache Spark to become ready')
+
+
+@when('zeppelin.started', 'client.notebook.registered')
+def register_notebook(client):
+    notebooks = client.unregistered_notebooks()
+    api = ZeppelinAPI()
+    id_map = unitdata.kv().get('zeppelin.notebooks.id_map', {})
+    for notebook in notebooks:
+        notebook_md5 = hashlib.md5(notebook.encode('utf8')).hexdigest()
+        notebook_id = api.import_notebook(notebook)
+        if notebook_id:
+            hookenv.log('Registered notebook: {} -> {}'.format(notebook_md5,
+                                                               notebook_id))
+            id_map[notebook_md5] = notebook_id
+            client.accept_notebook(notebook)
+        else:
+            hookenv.log('Rejected notebook: {}'.format(notebook_md5))
+            client.reject_notebook(notebook)
+    unitdata.kv().set('zeppelin.notebooks.id_map', id_map)
+
+
+@when('zeppelin.started', 'client.notebook.removed')
+def remove_notebook(client):
+    notebooks = client.unremoved_notebooks()
+    api = ZeppelinAPI()
+    id_map = unitdata.kv().get('zeppelin.notebooks.id_map', {})
+    for notebook in notebooks:
+        notebook_md5 = hashlib.md5(notebook.encode('utf8')).hexdigest()
+        notebook_id = id_map.get(notebook_md5, None)
+        if not notebook_id:
+            hookenv.log('Skipping removing unknown notebook: {}'.format(
+                notebook_md5))
+            continue
+        hookenv.log('Removing notebook: {} -> {}'.format(notebook_md5,
+                                                         notebook_id))
+        api.delete_notebook(notebook_id)
+        client.remove_notebook(notebook)
+        del id_map[notebook_md5]
+    unitdata.kv().set('zeppelin.notebooks.id_map', id_map)
+
+
+@when('zeppelin.started', 'client.interpreter.change')
+def modify_interpreter(client):
+    zepp = Zeppelin.get()
+    api = ZeppelinAPI()
+    for interpreter in client.interpreter_changes():
+        try:
+            api.modify_interpreter(interpreter['name'],
```

Comment:
Same as above, need to make sure zepp is actually running before calling this, else the `requests.get` to localhost:9090 will fail.